src/fromager/sources.py

```diff
@@ -369,32 +369,40 @@ def prepare_source(
     version: Version,
 ) -> pathlib.Path:
     logger.info(f"{req.name}: preparing source for {req} from {source_filename}")
-    source_root_dir = overrides.find_and_invoke(
+    prepare_source_details = overrides.find_and_invoke(
         req.name,
         "prepare_source",
-        _default_prepare_source,
+        default_prepare_source,
         ctx=ctx,
         req=req,
         source_filename=source_filename,
         version=version,
     )
+    if len(prepare_source_details) == 1:
```

Comment:
If there's only 1 return value, it won't be a tuple, it'll just be that value. I think you'll need to use `isinstance()` in the checks.

Reply:
ahh yes